### PR TITLE
feat: impl cluster meta storage

### LINKED ISSUE #14: feat: define MetaStorage interface
**Description**

define MetaStorage interface to persist ceresdb cluster info

**Proposal**
ceresdb cluster info refer to https://github.com/CeresDB/ceresdbproto/blob/main/protos/meta/cluster.proto

**Additional context**

### PR DESCRIPTION
**Pull request by chunshao90**
# Which issue does this PR close?

Closes #14 

# Rationale for this change
 
 

# What changes are included in this PR?

# Are there any user-facing changes?

# How does this change test

### CODE REVIEW

File: server/storage/storage_impl.go [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+	"time"
+
+	"github.com/CeresDB/ceresdbproto/pkg/metapb"
+	clientv3 "go.etcd.io/etcd/client/v3"
+	"google.golang.org/protobuf/proto"
+)
+
+type Options struct {
+	// maxScanLimit is the max limit of the number of keys in a scan.
+	maxScanLimit int
```
Comment (ShiKaiWi) [inline server/storage/storage_impl.go:17]: Should these fields be public as it is an `Options`?

File: server/storage/storage_impl.go [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+	requestTimeout time.Duration
+}
+
+// MetaStorageImpl is the base underlying storage endpoint for all other upper
+// specific storage backends. It should define some common storage interfaces and operations,
+// which provIDes the default implementations for all kinds of storages.
+type MetaStorageImpl struct {
+	KV
+
+	maxScanLimit int
```
Comment (ShiKaiWi) [inline server/storage/storage_impl.go:30]: How about just keep the whole `Options`.

File: server/storage/storage_impl.go [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+}
+
+func (s *MetaStorageImpl) GetSchemas(ctx context.Context, clusterID uint32, schemas []*metapb.Schema) error {
+	nextID := uint32(0)
+	endKey := schemaPath(clusterID, math.MaxUint32)
+
+	rangeLimit := s.maxScanLimit
+	for {
+		startKey := schemaPath(clusterID, nextID)
+		_, res, err := s.Scan(startKey, endKey, rangeLimit)
```
Comment (ShiKaiWi) [inline server/storage/storage_impl.go:73]: Maybe we should add a context as the input to `KV.Scan` for setting a reasonable timeout.

File: server/storage/key_path.go [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+	cluster = v1 + "cluster"
+	schema  = "schema"
+)
+
+// schemaPath returns the schema meta info key path with the given region ID.
+// example:
+// cluster 1: v1/cluster/1/schema/1 -> ceresmeta.Schema
+//            v1/cluster/1/schema/2 -> ceresmeta.Schema
+//            v1/cluster/1/schema/3 -> ceresmeta.Schema
+func schemaPath(clusterId uint32, schemaId uint32) string {
```
[the remark this answers is not in the document]
Comment (chunshao90) [reply server/storage/key_path.go:21]: schemaKey

**Review by ShiKaiWi — DISMISSED** [review]
LGTM

**Pull request merged**